DSharpPlus/Net/Rest/DiscordApiClient.cs

```diff
@@ -2031,7 +2033,21 @@ internal async Task<DiscordRole> CreateGuildRoleAsync(ulong guild_id, string nam
             var bucket = this.Rest.GetBucket(RestRequestMethod.POST, route, new { guild_id }, out var path);
 
             var url = Utilities.GetApiUriFor(path);
-            var res = await this.DoRequestAsync(this.Discord, bucket, url, RestRequestMethod.POST, route, headers, DiscordJson.SerializeObject(pld)).ConfigureAwait(false);
+
+            RestResponse res;
+            if (icon is null)
+            {
+                res = await this.DoRequestAsync(this.Discord, bucket, url, RestRequestMethod.POST, route, headers, DiscordJson.SerializeObject(pld)).ConfigureAwait(false);
+            }
+            else
+            {
+                var values = new Dictionary<string, string>()
+                {
+                    ["payload_json"] = DiscordJson.SerializeObject(pld)
+                };
+
+                res = await this.DoMultipartAsync(this.Discord, bucket, url, RestRequestMethod.POST, route, headers, values, new[] { icon });
+            }
```

Comment:
This is incorrect.
The icon must be transmitted as base64

Reply:
https://github.com/discord/discord-api-docs/blob/master/docs/Reference.md#image-data

Reply:
I'm aware and fixing this.